src/genericComponents/Button.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import styled from 'styled-components';
+// import { colors, fonts } from '../theme/styleVars';
+
+const ButtonPrimary = (props: any) => {
+  const { onClick, text, variant } = props;
+
+  return (
+    <CAGButton
+      className={`${variant}-class`}
+      onClick={onClick}
+      variant={variant}
+    >
+      {text}
+    </CAGButton>
+  );
+};
+
+const CAGButton = styled(Button)`
```

Comment:
Your style for the block here won't work because of how you have your curly braces (the classes won't count). So what you need is:
```
const CAGButton = styled(Button)`
  border-radius: 25px;
  box-shadow: 2px 2px 20px rgba(0, 0, 0, 29);
  character-spacing: 70;
  font-family: {fonts.montserrat};
  font-size: 12px;
  font-weight: bold;
  line-spacing: 15;
  transform: uppercase;

  &.primary-class { ...stuff here }
  &.secondary-class { ...stuff here }
`;